core/shared/src/test/scala/org/typelevel/log4cats/SamStructuredLoggerTest.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats
+
+import cats.effect.IO
+import munit.CatsEffectSuite
+
+class SamStructuredLoggerTest extends CatsEffectSuite {
+
+  test("SamStructuredLogger should handle context with multiple entries") {
+    var capturedLogs: List[(KernelLogLevel, Log.Builder[String] => Log.Builder[String])] = Nil
+
+    val testKernel = new LoggerKernel[IO, String] {
```

Comment:
issue: Instead of inlining this with a `var`, it'd be better to rework the existing test loggers to use `LoggerKernel`